src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/CSharpOnTypeFormattingPass.cs

```diff
@@ -41,11 +44,12 @@ internal class CSharpOnTypeFormattingPass : CSharpFormattingPassBase
         }
 
         _logger = loggerFactory.CreateLogger<CSharpOnTypeFormattingPass>();
+        _optionsMonitor = optionsMonitor;
     }
 
     public async override Task<FormattingResult> ExecuteAsync(FormattingContext context, FormattingResult result, CancellationToken cancellationToken)
     {
-        if (!context.IsFormatOnType || result.Kind != RazorLanguageKind.CSharp)
+        if (!context.IsFormatOnType || result.Kind != RazorLanguageKind.CSharp || !_optionsMonitor.CurrentValue.FormatOnType)
```

Comment:
Good catch, I'll add to that as well

Reply:
hmm.. just a random thought.. does this code path get hit when a C# code action is executed? Or when doing `override` completion?

Reply:
I would hope we have tests for that 🤞. That said, looks like MEF cache is causing issues in CI runs right now so I'll have to take a look next week.

Reply:
Yeah, but unless you updated all of those tests to also run with the option turned off, they won't help much 😛

I'm sure the features still work, I just don't know how mangled things will be afterwards. In general I'm curious to see what the experience with this off is like, because I could definitely see a future where we control C# and Html on type formatting separately.